WearTilesKotlin/app/src/main/java/com/example/wear/tiles/TilesApplication.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.wear.tiles
+
+import android.app.Application
+import coil.ImageLoader
+import coil.ImageLoaderFactory
+import com.example.wear.tiles.di.AppContainer
+
+class TilesApplication : Application(), ImageLoaderFactory {
+    internal lateinit var appContainer: AppContainer
+
+    override fun onCreate() {
+        super.onCreate()
+
+        appContainer = AppContainer(this)
+    }
+
+    override fun onTerminate() {
```

Comment:
>This method is for use in emulated process environments. It will never be called on a production Android device, where processes are removed by simply killing them; no user code (including this callback) is executed when doing so.

Is this intentional? I've never used/seen it before.